pkg/installmanager/installmanager.go

```diff
@@ -1810,3 +1820,62 @@ func (m *InstallManager) overrideCredentialsModeOnInstallConfig(credentialsMode
 
 	return nil
 }
+
+func (m *InstallManager) patchAzureOverrideCreds() error {
+	overrideCredsPath := filepath.Join(m.WorkDir, "manifests", overrideCredsYAML)
+	overrideCredsBytes, err := ioutil.ReadFile(overrideCredsPath)
+	if err != nil {
+		return errors.Wrap(err, "error reading 99_cloud-creds-secret.yaml")
+	}
+	c, err := yamlutils.Decode(overrideCredsBytes)
+	if err != nil {
+		return errors.Wrap(err, "unable to decode 99_cloud-creds-secret.yaml content")
+	}
+	if isAzureCreds, _ := yamlutils.Test(c, "/metadata/name", "azure-credentials"); !isAzureCreds {
+		return errors.Wrap(err, "wrong input, secret name is not azure-credentials in 99_cloud-creds-secret.yaml")
+	}
+	region, err := exec.Command("bash", "-c", fmt.Sprintf("grep %s %s |awk -F ': ' '{print $2}'", "region", filepath.Join(m.WorkDir, "manifests", clusterConfigYAML))).Output()
+	if err != nil {
+		return errors.Wrap(err, "error to grep region from manifests")
+	}
+	infraName, err := exec.Command("bash", "-c", fmt.Sprintf("grep %s %s |awk -F ': ' '{print $2}'", "infrastructureName", filepath.Join(m.WorkDir, "manifests", clusterInfraConfigYAML))).Output()
+	if err != nil {
+		return errors.Wrap(err, "error to grep infrastructureName from manifests")
+	}
+	resourceGroupName, err := exec.Command("bash", "-c", fmt.Sprintf("grep %s %s |awk -F ': ' '{print $2}'", "resourceGroupName", filepath.Join(m.WorkDir, "manifests", clusterInfraConfigYAML))).Output()
+	if err != nil {
+		return errors.Wrap(err, "error to grep resourceGroupName from manifests")
+	}
+	regionBase64 := interface{}(base64.RawStdEncoding.EncodeToString(region))
+	infraNameBase64 := interface{}(base64.RawStdEncoding.EncodeToString(infraName))
+	resourceGroupNameBase64 := interface{}(base64.RawStdEncoding.EncodeToString(resourceGroupName))
+	ops := yamlpatch.Patch{
```

Comment:
Good question!
Accept is not good, since it may cause cluster resources are created in two resource groups (installer create one and image-registry uses the inputted resource group).

I am not sure if replace is good for user since user may not notice this change at all.

I prefer to report error if azure_resource_prefix/azure_resourcegroup are already supplied, region can be replaced, WDYT?

Reply:
I would prefer consistent behavior across all fields; but I agree with your reasoning. So let's report an error if any field is already specified *and* the value is different from whatever we're laying down.

Reply:
Sounds good.

Reply:
For simplicity, I just return error if any of azure_resource_prefix/azure_resourcegroup already exists, because the infra Id was generated randomly when running "openshift-install create manifests", so user is not able to configure it in advance. If need to handle the exist case, will make the later yamlpatch logic complicated.

Reply:
I'm fine with that for those two fields. But now I think I see your issue with the region. Since that field doesn't rely on the infra ID (or anything generated, right?) it would be totally reasonable for the user to include it in their 99_cloud-creds-secret.yaml file, and we shouldn't error if that's the case -- unless it doesn't match what they put in the cd.spec.

So, sorry to change my mind, I think it's worthwhile to split that out into a separate condition.

Reply:
OK. I changed the logic as conclusion.
If infraName/resourceGroupName already exist, report error
If region exists and different to the one in cd.spec, report error